collector/src/descriptor/kind/server_descriptor.rs

```diff
@@ -63,9 +73,24 @@ impl ServerDescriptor {
                     or_port_v6: address.map(str::parse::<SocketAddrV6>).transpose()?
                         .as_ref().map(SocketAddrV6::port),
                 },
+                cert("identity-ed25519") [certif] => {
+                    identity_ed25519: certif.to_owned(),
+                },
+                uniq("master-key-ed25519") [key] => {
+                    master_key_ed25519: key.to_owned(),
+                },
                 uniq("platform") [] => {
                     platform: rest.join(" "),
                 },
+                opt("proto") [] => {
+                    // TODO should reject when split_once fail
+                    proto: rest.map(|r|
+                                    r.iter()
+                                    .filter_map(|v| v.split_once('='))
+                                    .map(|(k,v)| (k.to_owned(), v.to_owned()))
+                                    .collect()
+                                ).unwrap_or_default(),
```

Comment:
if you want to fix the todo in this pr (if you don't, feel free to not), you could do something along the line of 
```rust
rest.map(|r|
    r.iter()
     .map(|v| 
         v.split_once('=')
          .map(|(k,v)| (k.to_owned(), v.to_owned()))
          .ok_or(Error::Something)
     )
     .collect::<Result<Vec<_>, _>()
)
.transpose()? // convert Option<Result> to Result<Option>, propagate error if Result is Err
.unwrap_or_default(),